### DISCUSSION TITLE
OnDelete not triggering Middleware

### DISCUSSION THREAD

**Discussion by Rykuno** [prisma/prisma#10655] [Q&A] [answered]
I'm curious to the interaction between utilizing the `onDelete: Cascade` from within the `referentialIntegrity` plugin and how it interfaces w/ middleware. 

I expected once a `Post` which has a relation to `Image` was deleted, the hook would be called for `Post` and `Image`, but instead it appears to only trigger for `Post`.

Is this intentional that a referenced object being deletion from `onDelete` is not supposed to go through the middleware? 

Here is a dumbed down model/relation for reference

```prisma
model Post {
  id       String  @id @default(uuid())
  title    String?
  images   Image[]
}

model Image {
  id   String @id @default(uuid())
  
  post   Post   @relation(fields: [postId], references: [id], onDelete: Cascade)
  postId String

  createdAt DateTime @default(now())
}
```

The below hook does not get run for the Image model in this case(A deletion of Post).

```js
prisma.$use(async (params, next) => {
  if (params.model !== "Image") return next(params)

  const result = await next(params)
  return result
})
```

**Comment by austincrim** [comment, marked as answer]
Hey @Rykuno 👋 !

This is a great question that I briefly discussed with our engineering teams. Referential actions are actually defined and executed at the *database* level, not at the Prisma Client level. This is why the Prisma Client middleware doesn't pick them up.

It's something we should mention in our docs, so I'll create an issue in our docs repo to make sure we update those pages.

If this is blocking a use case for you, could you describe what you're trying to accomplish and maybe we can create a feature request or find a workaround?

Thanks so much!

**Reply by Rykuno** [reply]
It's currently not blocking. 

Essentially in our use case we have many models w/ relation to files/images stored in S3. We were just hoping to utilize the middleware to handle the deletion in a single area instead of duplicating logic.

I reread the docs after you mentioned them being executed on the database level and it does mention it https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions

Three of us missed that key point even after reading the docs but we might be outliers who missed that dropdown of `Referential actions on the database level`. 

Thanks for the help!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
